dashboard/app/models/teacher_feedback.rb

```diff
@@ -29,10 +29,15 @@ class TeacherFeedback < ApplicationRecord
   acts_as_paranoid # use deleted_at column instead of deleting rows
   validates_presence_of :student_id, :script_id, :level_id, :teacher_id, :script_level_id, unless: :deleted?
   belongs_to :student, class_name: 'User'
+
+  # despite newer teacher feedbacks being explicitly associated with a section,
+  # this association is necessary because we did not record section associations for older teacher feedbacks
   has_many :student_sections, class_name: 'Section', through: :student, source: 'sections_as_student'
+
   belongs_to :script
   belongs_to :level
   belongs_to :script_level
+  belongs_to :section
```

Comment:
I think this could be confusing long-term with two different ways to get the section in this model.  Do we think that the new section_id column will ultimately become the source of truth for the associated section OR is `student_sections` above the source of truth and section_id is just for convenience?  If the former, then I think we need to make that more explicit, possibly by creating a method in this this model that makes the fallback explicit.  If it's the latter, then maybe we need to name it something other than section_id?

Reply:
It seems a lot cleaner to me if we could have section ID be the source of truth for the associated section, but yes, there's the problem of historical data, so it's not possible right now for the new column to be that source of truth. We _could_ push to do the backfill I mentioned in [this comment](https://github.com/code-dot-org/code-dot-org/pull/39115#issuecomment-780907871).

I thought a bit about how combining the old and new associations, to your point we could have some instance method like:

```
def legacy_compliant_sections
  section || get_me_the_most_relevant_section(student_sections)
end
```

But [the way we use](https://github.com/code-dot-org/code-dot-org/blob/5397cfe7056b8a2273460cdf1c12f80ed7bbf3ec/dashboard/app/models/teacher_feedback.rb#L91) the `student_sections` association (at least currently), we need some association that is defined as part of the model definition (like a `has_one`) that combines the two associations, which I couldn't figure out how to do.

All that to say I kind of came to the conclusion we'd need to do the backfill if we want to use the section association, and fully deprecate the existing `student_sections` association.

What would you name it if not section ID?

Reply:
i just want to point out that if this is primarily being added for analytics at this point, doing that backfill might be valuable not only from a tech-debt perspective, but also from a feature-value perspective. presumably a successful backfill would let our data team start looking at this retroactively, rather than waiting to see data going forward, right?

it sounds to me like the backfill would allow for a cleaner technical implementation with less tech debt and would therefore be ideal, budget allowing. so if it's true that it would also provide us with historical data on this metric i would float the idea with amanda/clare of expanding the scope slightly for this double benefit.

Reply:
That is reasonable, let me check with them...

Reply:
Asked in Slack -- sounds like she's not up for this for now.